lightning/cogs/automod/ui.py

```diff
@@ -191,3 +203,338 @@ async def warn_ban(self, itx: discord.Interaction, button: discord.ui.Button):
+
+        if role is None:
+            await itx.response.send_message("Somehow you clicked this button without setting a role first!",
+                                            ephemeral=True)
+            return
+
+        view = GatekeeperChannelView(role, channel,
+                                     author_id=itx.user.id, delete_message_after=True)
+        await itx.response.send_message(content='Select a channel from the select menu below',
+                                        view=view, ephemeral=True)
+        await view.wait()
+        self.invalidate_gatekeeper_cache()
+        await self.update(interaction=itx)
+
+    @discord.ui.button(label="Send verification message", style=discord.ButtonStyle.blurple)
+    async def send_verification_message(self, itx: discord.Interaction[LightningBot], button: discord.ui.Button):
+        self.gatekeeper = await self.ctx.cog.get_gatekeeper_config(itx.guild_id)
+        if self.gatekeeper is None:
+            await itx.response.send_message("Somehow your gatekeeper isn't setup correctly!")
+            return
+
+        embed = discord.Embed(title="Verification Required",
+                              description="This server requires you to verify yourself before you can talk!")
+
+        ch = self.gatekeeper.verification_channel
+
+        if ch is None:
+            await itx.response.send_message("Please set a verification channel before setting this up!", ephemeral=True)
+            return
+
+        view = discord.ui.View(timeout=None)
+        view.add_item(GatekeeperVerificationButton(self.gatekeeper))
+        await ch.send(embed=embed, view=view)
+
+    @discord.ui.button(label="Disable", style=discord.ButtonStyle.red)
+    async def disable_gatekeeper(self, itx: discord.Interaction[LightningBot], button: discord.ui.Button):
+        self.gatekeeper = await self.ctx.cog.get_gatekeeper_config(itx.guild_id)  # type: ignore
+        if self.gatekeeper is None:
+            await itx.response.send_message("Somehow your gatekeeper isn't setup correctly!")
+            return
+
+        if button.style is discord.ButtonStyle.green:
+            await self.gatekeeper.enable()
+            await itx.response.send_message("Enabled the gatekeeper. Every new member will be required to verify.",
+                                            ephemeral=True)
+            # self.gatekeeper.active_since = datetime.now()
+        else:
+            await self.gatekeeper.disable()
+            query = "UPDATE guild_gatekeeper_config SET active='f' WHERE guild_id=$1;"
+            await itx.client.pool.execute(query, itx.guild_id)
+            await itx.response.send_message("Disabled the gatekeeper. Removing members from the queue could take some "
+                                            "time!", ephemeral=True)
+
+        await self.update(interaction=itx)
+
+
+class GatekeeperVerificationButton(discord.ui.DynamicItem[discord.ui.Button],
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Verify `GatekeeperVerificationButton` correctly handles user verification.

Testing is needed to ensure that `GatekeeperVerificationButton` properly verifies users and interacts with the gatekeeper configuration to update member status.